2_Scope_Closures/practice/closure_part3_yujo.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+function formatTotal(display) {
+  if (Number.isFinite(display)) {
+    // constrain display to max 11 chars
+    let maxDigits = 11;
+    // reserve space for "e+" notation?
+    if (Math.abs(display) > 99999999999) {
+      maxDigits -= 6;
+    }
+    // reserve space for "-"?
+    if (display < 0) {
+      maxDigits--;
+    }
+
+    // whole number?
+    if (Number.isInteger(display)) {
+      display = display.toPrecision(maxDigits).replace(/\.0+$/, "");
+    }
+    // decimal
+    else {
+      // reserve space for "."
+      maxDigits--;
+      // reserve space for leading "0"?
+      if (Math.abs(display) >= 0 && Math.abs(display) < 1) {
+        maxDigits--;
+      }
+      display = display.toPrecision(maxDigits).replace(/0+$/, "");
+    }
+  } else {
+    display = "ERR";
+  }
+  return display;
+}
+
+function useCalc(calc, keys) {
+  return [...keys].reduce(function showDisplay(display, key) {
+    var ret = String(calc(key));
+    return display + (ret != "" && key == "=" ? "=" : "") + ret;
+  }, "");
+}
+
+function calculator() {
+  let totalValue = 0;
+  let currentVaule = "";
+  let currentOp = "";
+  let restart = false;
+
+  const operators = ["+", "-", "*", "/", "%"];
+
+  function isNum(key) {
+    return Number(key) ? true : false;
```

Comment:
```suggestion
    return !isNaN(Number(key)) ? true : false;
```
해당 조건 처리에서 인자 값으로 '0' 이 들어 올 경우엔 number type으로 치환된 0이 falsy 값을 갖기 때문에 버그 발생 요지가 있습니다! 조금 더 엄밀한 처리를 위해 NaN 여부를 확인하시는 걸 추천드립니다